### PR TITLE
Clean up pseudo-private member usage in `passes/sat/expose.cc`.

### PR DESCRIPTION
**Pull request by boqwxp**

### CODE REVIEW

File: passes/sat/expose.cc [hunk 1]
```diff
@@ -314,26 +314,26 @@ struct ExposePass : public Pass {
...
-					first_module = mod_it.second;
+					first_module = mod;
 				} else {
 					std::set<RTLIL::IdString> new_shared_dff_wires;
 					for (auto &it : shared_dff_wires) {
-						if (!dff_dq_maps[mod_it.second].count(it))
+						if (!dff_dq_maps[mod].count(it))
 							continue;
-						if (!compare_wires(first_module->wires_.at(it), mod_it.second->wires_.at(it)))
+						if (!compare_wires(first_module->wires_.at(it), mod->wires_.at(it)))
```
Comment (nakengelhardt) [inline passes/sat/expose.cc]: This could also be `first_module->wire(it)` and `mod->wire(it)`.

File: passes/sat/expose.cc [hunk 2]
```diff
@@ -314,26 +314,26 @@ struct ExposePass : public Pass {
 			RTLIL::Module *first_module = NULL;
 			std::set<RTLIL::IdString> shared_dff_wires;
 
-			for (auto &mod_it : design->modules_)
+			for (auto mod : design->modules())
 			{
-				if (!design->selected(mod_it.second))
+				if (!design->selected(mod))
 					continue;
```
Comment (nakengelhardt) [inline passes/sat/expose.cc]: This is equivalent to `design->selected_modules()`

File: passes/sat/expose.cc [hunk 3]
```diff
@@ -364,10 +364,8 @@ struct ExposePass : public Pass {
 		{
 			RTLIL::Module *first_module = NULL;
 
-			for (auto &mod_it : design->modules_)
+			for (auto module : design->modules())
 			{
-				RTLIL::Module *module = mod_it.second;
-
 				if (!design->selected(module))
 					continue;
```
Comment (nakengelhardt) [inline passes/sat/expose.cc]: This is equivalent to `design->selected_modules()` too

File: passes/sat/expose.cc [hunk 4]
```diff
@@ -101,9 +101,9 @@ void create_dff_dq_map(std::map<RTLIL::IdString, dff_map_info_t> &map, RTLIL::De
 	std::map<RTLIL::SigBit, dff_map_bit_info_t> bit_info;
 	SigMap sigmap(module);
 
-	for (auto &it : module->cells_)
+	for (auto cell : module->cells())
 	{
-		if (!design->selected(module, it.second))
+		if (!design->selected(module, cell))
 			continue;
```
Comment (nakengelhardt) [inline passes/sat/expose.cc]: This is equivalent to `module->selected_cells()`

File: passes/sat/expose.cc [hunk 5]
```diff
@@ -446,10 +444,8 @@ struct ExposePass : public Pass {
 			}
 		}
 
-		for (auto &mod_it : design->modules_)
+		for (auto module : design->modules())
 		{
-			RTLIL::Module *module = mod_it.second;
-
 			if (!design->selected(module))
 				continue;
```
Comment (nakengelhardt) [inline passes/sat/expose.cc]: This is `design->selected_modules()` too

**Comment by boqwxp** [comment]
Nice!  Good eye.

**Pull request merged**